hexl/fft/inv-fft-avx512.cpp

```diff
@@ -0,0 +1,367 @@
+        W_1C_intrlvd[2], W_1C_intrlvd[2], W_1C_intrlvd[0], W_1C_intrlvd[0]);
+    __m512d v_W_imag = _mm512_set_pd(
+        W_1C_intrlvd[3], W_1C_intrlvd[3], W_1C_intrlvd[1], W_1C_intrlvd[1],
+        W_1C_intrlvd[3], W_1C_intrlvd[3], W_1C_intrlvd[1], W_1C_intrlvd[1]);
+
+    W_1C_intrlvd += 4;
+
+    ComplexInvButterfly(&v_X_real, &v_X_imag, &v_Y_real, &v_Y_imag, v_W_real,
+                        v_W_imag);
+
+    ComplexWriteInvInterleavedT4(v_X_real, v_Y_real, v_X_pt_real);
+    ComplexWriteInvInterleavedT4(v_X_imag, v_Y_imag, v_X_pt_imag);
+
+    offset += 32;
+  }
+}
+
+void ComplexInvT8(double_t* operand_8C_intrlvd, const double_t* W_1C_intrlvd,
+                  uint64_t gap, uint64_t m) {
+  size_t offset = 0;
+
+  HEXL_LOOP_UNROLL_4
+  for (size_t i = 0; i < (m >> 1); i++) {
+    // Referencing operand
+    double_t* X_real = operand_8C_intrlvd + offset;
+    double_t* X_imag = operand_8C_intrlvd + 8 + offset;
+
+    double_t* Y_real = X_real + gap;
+    double_t* Y_imag = X_imag + gap;
+
+    __m512d* v_X_pt_real = reinterpret_cast<__m512d*>(X_real);
+    __m512d* v_X_pt_imag = reinterpret_cast<__m512d*>(X_imag);
+
+    __m512d* v_Y_pt_real = reinterpret_cast<__m512d*>(Y_real);
+    __m512d* v_Y_pt_imag = reinterpret_cast<__m512d*>(Y_imag);
+
+    // Weights
+    __m512d v_W_real = _mm512_set1_pd(*W_1C_intrlvd++);
+    __m512d v_W_imag = _mm512_set1_pd(*W_1C_intrlvd++);
+
+    // assume 8 | t
+    for (size_t j = 0; j < gap; j += 16) {
+      __m512d v_X_real = _mm512_loadu_pd(v_X_pt_real);
+      __m512d v_X_imag = _mm512_loadu_pd(v_X_pt_imag);
+
+      __m512d v_Y_real = _mm512_loadu_pd(v_Y_pt_real);
+      __m512d v_Y_imag = _mm512_loadu_pd(v_Y_pt_imag);
+
+      ComplexInvButterfly(&v_X_real, &v_X_imag, &v_Y_real, &v_Y_imag, v_W_real,
+                          v_W_imag);
+
+      _mm512_storeu_pd(v_X_pt_real, v_X_real);
+      _mm512_storeu_pd(v_X_pt_imag, v_X_imag);
+
+      _mm512_storeu_pd(v_Y_pt_real, v_Y_real);
+      _mm512_storeu_pd(v_Y_pt_imag, v_Y_imag);
+
+      // Increase operand & result pointers
+      v_X_pt_real += 2;
+      v_X_pt_imag += 2;
+      v_Y_pt_real += 2;
+      v_Y_pt_imag += 2;
+    }
+    offset += (gap << 1);
+  }
+}
+
+// Takes operand as 8 complex interleaved: This is 8 real parts followed by
+// its 8 imaginary parts.
+// Returns operand as 1 complex interleaved: One real part followed by its
+// imaginary part.
+void ComplexFinalInvT8(double_t* operand_8C_intrlvd,
+                       const double_t* W_1C_intrlvd, uint64_t gap, uint64_t m,
+                       const double_t* scalar = nullptr) {
+  size_t offset = 0;
+
+  __m512d v_scalar;
+  if (scalar != nullptr) {
+    v_scalar = _mm512_set1_pd(*scalar);
+  }
+
+  HEXL_LOOP_UNROLL_4
+  for (size_t i = 0; i < (m >> 1); i++) {
+    // Referencing operand
+    double_t* X_real = operand_8C_intrlvd + offset;
+    double_t* X_imag = operand_8C_intrlvd + 8 + offset;
```

Comment:
Similar suggestion. Or you could have x_imag = (*x_real) + 8;
```suggestion
    double_t* X_imag = operand_8C_intrlvd + 8 + offset;
```

Reply:
I will be using double_t* X_r_imag = X_r_real + 8; everywhere